compiler/main/config.cpp

```diff
@@ -68,40 +85,53 @@ void checkConfigs() {
 // the internal scope-less BlockStmt.
 
 void parseCmdLineConfig(const char* name, const char* value) {
-  // Generate a C-string for a nominal Chapel assignment statement
-  const char* stmtText = (value[0] != '\0') ? astr("dummyConfig=", value, ";") : astr("dummyConfig=true;");
-  const char* parseFn  = astr("Command-line arg (", name, ")");
-  const char* parseMsg = astr("parsing '", value, "'");
 
-  // Invoke the parser to generate AST
-  BlockStmt*  stmt     = parseString(stmtText, parseFn, parseMsg);
+  // unfortunately this is parsed in the order from the command line, so for
+  // it to work, --dyno must come before -sConfigVar=Val or it will not try
+  // to use the dyno parser for command line input
+  if (fDynoCompilerLibrary) {
+    // emplace the record for dyno parser
```

Comment:
but it doesn't emplace, it copies. If you want it to do something I would call "emplace" it would be

```
gDynoParams.push_back(std::make_pair(std::string(symbol), std::string(val)));
```

Since we should not be concerned about the performance here, you could alternatively remove "emplace" from the comment :)